core/models/machine.py

```diff
@@ -319,9 +353,7 @@ def get_or_create_provider_machine(image_id, machine_name,
     if not version:
         version = get_version_for_machine(provider_uuid, image_id, fuzzy=True)
     if not version:
-        version = create_app_version(app, "1.0", provider_machine_id=image_id)
-    #TODO: fuzzy=True returns a list, but call comes through as a .get()?
-    #      this line will cover that edge-case.
+        version = create_app_version(app, version_name, provider_machine_id=image_id)
```

Comment:
This works as expected:
```
if version passed in, don't attempt to retrieve the version
if no version was passed in *and* attempt to retrieve version was empty, create a version.
```
To make this clearer, i will tab-over the second if statement.